index.go

```diff
@@ -4,8 +4,25 @@ import (
     "fmt"
     "log"
     "net/http"
+    "os"
+    "strconv"
 )
 
+// get the current listening address or fail if input is not correct
+func determineListenAddress() (string, error) {
+  portStr := os.Getenv("PORT")
+  
+  if portStr == "" {
+    return "", fmt.Errorf("$PORT not set")
+  }
+  
+  port, err := strconv.Atoi(portStr)
+  if err == nil {
+    return "", fmt.Errorf("$PORT not a valid integer")
```

Comment:
We could also verify that the Port value is in the range 1024 – 49151, but seems unnecessary

Reply:
ya honestly i felt a bit overboard even validating that it was an int. Maybe i should remove the code. I just feel weird with this arbitrary value set in the env.

Reply:
why does it make you feel weird?